File: Aquatrack-frontend/src/services/residentHouseholdService.js
import api from "../api/axios";

const BASE_URL = "/resident";

// ==========================================
// Resident Household Service
// ==========================================

const residentHouseholdService = {

    /**
     * Fetch logged-in resident household details.
     *
     * GET /api/resident/my-household
     */
    async getMyHousehold() {

        const response = await api.get(`${BASE_URL}/my-household`);

        return response.data.data;

    },

};

export default residentHouseholdService;

File: AquaTrack/src/main/java/com/aquatrack/service/ResidentHouseholdService.java
package com.aquatrack.service;

import com.aquatrack.dto.resident.ResidentHouseholdResponseDto;

public interface ResidentHouseholdService {

    // ==========================================
    // My Household
    // ==========================================

    /**
     * Returns the household details of the
     * currently logged-in resident.
     *
     * @return Resident household details
     */
    ResidentHouseholdResponseDto getMyHousehold();

}